src/compiler/crystal/macros.cr

```diff
@@ -1297,6 +1297,12 @@ module Crystal::Macros
     # returns a `NilLiteral`.
     def resolve? : ASTNode | NilLiteral
     end
+
+    # Returns this path inside an array litera or Union.
```

Comment:
```suggestion
    # Returns this path inside an array literal or Union.
```

Reply:
Thanks!

Reply:
If you squash it to the actual commit we can merge the PR without squashing it if you want.

Reply:
I don't know how to do that (my git knowledge is a bit limited) but I think we shouldn't worry too much about the commit history. A typo fix is a typo fix and I think it's fine if that appears in a separate commit.

Reply:
Yes, please just merge this without squashing.
But if there are more followups, manual squashing may become more relevant.

Then you can do `git rebase -i`, move the line of "fix typo" right under its relevant commit and also change it to be "fixup". And `git push --force`...

Reply:
Do I need to do git rebase -I master to fix this? I'm happy to learn.

Reply:
`$ git rebase -i master`

once the editor open move the last line 1 up so 2a4e6b6b3 is squashed with 746cc33f8.

```
pick f85222a82 Macros: add Union#resolve and Union#resolve?
pick d0562bbd2 Macros: add TypeNode#resolve and TypeNode#resolve?
pick 746cc33f8 Macros: add Path#types and Generic#types
s 2a4e6b6b3 Fix typo
pick 2cb3c1104 Macros: let TypeNode#union_types work for non-union types
```

save file
reword the commit message when the editor appears.

Or use GitKraken :-P (?)

Reply:
Wow, so easy! Thank you @oprypin and @bcardiff . This will save me a ton of time!